### PR TITLE
Code submission for FLIP fest 'Playground: Fix client-side errors in the playground frontend #58' - Milestone 1

### PR DESCRIPTION
**Pull request by hichana**
Closes: #163 #164 #165 #166 

## Description

This PR includes work done in PR #167. #167 resolves the main issues that were causing the cascade of console errors. As I wrote in that PR, and as encouraged in the FLIP issue, I refactored the code in order to help avoid similar issues in the future. This refactor was done as a part of FLIP fest issue [#58](https://github.com/onflow/flip-fest/issues/58). My basic thinking for this refactor was:

- There are 5 entries are added into the browser's history stack upon loading of the playground (as explained in my [0.5 submission](https://github.com/hichana/flip-fest/blob/submission/issue-%2358/milestone-%230.5/hichana/submission/issue-%2358/milestone-%230.5/hichana/%23%20milestone_readme.md).
- I saw a way to remove all of the redirects that were causing the multiple entries by moving the logic for fully loading the playground outside of the project provider (React Context API), which removes unnecessary redirects and helps reduce similar issues in the future. Appropriate data is then passed to the project provider.
- By reducing the number of redirects, the app is much more performant and results in a better experience for users.

______

For contributor use:

- [n/a ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/onflow/flow-playground/blob/master/CONTRIBUTING.md))
- [x ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x ] Code follows the [standards mentioned here](https://github.com/onflow/flow-playground/blob/master/CONTRIBUTING.md).
- [n/a ] Updated relevant documentation 
- [x ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: src/components/AccountList.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ const AccountList: React.FC = () => {
 
   const location = useLocation();
   const params = getParams(location.search)
-  const projectPath = isUUUID(project.id) ? project.id : "local"
+  const projectPath = isUUUID(project.id) ? project.id : "LOCAL-project"
```
Comment (MaxStalker) [inline src/components/AccountList.tsx:40]: Can we use all caps or all lowercase here? :)

File: src/components/Sidebar.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ const Sidebar: React.FC = () => {
 
   if (isLoading) return <p>Loading...</p>;
 
-  const projectPath = isUUUID(project.id) ? project.id : "local"
+  const projectPath = isUUUID(project.id) ? project.id : "LOCAL-project"
```
Comment (MaxStalker) [inline src/components/Sidebar.tsx:25]: Same caps/lowercase here :)
Maybe we can move constant somewhere and use it everywhere?

File: src/containers/Editor/index.tsx [hunk 3]
```diff
@@ -1,35 +1,140 @@
...
 const Playground: any = (props: any) => {
-  const params = getParams(props.location.search)
+
   const { projectId } = props;
+  
+  const params = getParams(props.location.search);
+  const { type, id } = params;
 
-  const isLocalProject = projectId === "local";
+  const isLocalProject = projectId === "LOCAL-project";
```
Comment (MaxStalker) [inline src/containers/Editor/index.tsx:20]: For example, we can export this value.

File: src/providers/Project/index.tsx [hunk 4]
```diff
@@ -257,6 +254,8 @@ export const ProjectProvider: React.FC<ProjectProviderProps> = ({
   };
 
   const getActiveEditor = (): ActiveEditor => {
+    // console.log("GET ACTIVE EDITOR ACTIVE:", active);
```
Comment (MaxStalker) [inline src/providers/Project/index.tsx:257]: We can remove `console.log` here.

**Review by MaxStalker — CHANGES_REQUESTED** [review]

**Comment by hichana** [comment]
Thanks for the feedback @MaxStalker. I made the revisions:
- removed the console.log
- added an exported constant for the local project id in 'url.ts', which is the same place where another helper is exported ('scriptTypes').
- made all places where the local project id is consumed import the local project id
- changed the other local ids in projectDefault.ts to also be lowercase, but did not add them to 'url.ts' as they are only used in the projectDefault.ts file

File: src/util/url.ts [hunk 5]
```diff
@@ -12,3 +12,5 @@ export const getParams = (url: string): any =>  {
 }
 
 export const scriptTypes = ["account", "tx", "script"]
+
+export const localProjectId = "local-project"
```
Comment (MaxStalker) [inline src/util/url.ts:16]: Can we use to all caps with underscore for const name (not the value), i.e. `LOCAL_PROJECT_ID`?
This way it should be easier to catch up that value is constant and defined elsewhere.

**Review by MaxStalker — CHANGES_REQUESTED** [review]

File: src/util/url.ts (hunk 5 above)
Comment (hichana) [reply src/util/url.ts]: Got it. This actually crossed my mind while making the changes but for some reason I didn't do it. Change made and pushed to this same branch/PR!

**Review by MaxStalker — APPROVED** [review]
Looking good! :)

**Review by 10thfloor — APPROVED** [review]

**Pull request merged**